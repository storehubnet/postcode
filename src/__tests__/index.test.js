/* eslint-env jest */

const postcode = require('../index');

describe('postcodejs:jest', () => {
    test('Func:postcode:String', () => {
        const res = postcode.postcode('MY', '96507');
        expect(res).toMatchSnapshot();
    });

    test('Func:postcode:Number', () => {
        const res = postcode.postcode('MY', 96507);
        expect(res).toMatchSnapshot();
    });

    test('Func:statePostcodes', () => {
        const res = postcode.statePostcodes('MY', 'Sarawak');
        expect(res).toMatchSnapshot();
    });

    test('Func:districtPostcodes', () => {
        const res = postcode.districtPostcodes('MY', 'Padang Terap');
        expect(res).toMatchSnapshot();
    });

    test('Func:cityPostcodes', () => {
        const res = postcode.cityPostcodes('MY', 'Kangar');
        expect(res).toMatchSnapshot();
    });
});