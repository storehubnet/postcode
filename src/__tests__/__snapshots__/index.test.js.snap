// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcodejs:jest Func:cityPostcodes 1`] = `
Array [
  "1000",
  "1007",
  "1009",
  "1500",
  "1502",
  "1503",
  "1504",
  "1505",
  "1506",
  "1508",
  "1512",
  "1514",
  "1516",
  "1517",
  "1518",
  "1524",
  "1529",
  "1532",
  "1538",
  "1540",
  "1546",
  "1550",
  "1551",
  "1556",
  "1560",
  "1564",
  "1570",
  "1572",
  "1576",
  "1578",
  "1582",
  "1586",
  "1590",
  "1592",
  "1594",
  "1596",
  "1598",
  "1600",
  "1604",
  "1606",
  "1608",
  "1609",
  "1610",
  "1612",
  "1614",
  "1620",
  "1622",
  "1626",
  "1628",
  "1630",
  "1632",
  "1634",
  "1644",
  "1646",
  "1648",
  "1660",
  "1664",
  "1670",
  "1672",
  "1673",
  "1674",
  "1676",
  "1680",
  "1694",
  "2400",
  "2450",
  "2500",
]
`;

exports[`postcodejs:jest Func:districtPostcodes 1`] = `
Array [
  "93690",
  "93694",
  "93700",
  "93702",
  "93704",
  "93706",
  "93708",
  "93710",
  "93712",
  "93714",
  "93716",
  "93718",
  "93720",
  "93722",
  "93724",
  "93726",
  "93728",
  "93730",
  "93732",
  "93734",
  "93736",
  "93738",
  "93740",
  "93742",
  "93744",
  "93746",
  "93748",
  "93750",
  "93752",
  "93754",
  "93756",
  "93758",
  "93760",
  "93762",
  "93764",
  "93900",
  "93902",
  "93904",
  "93906",
  "93908",
  "93910",
  "93912",
  "93914",
  "93916",
  "93990",
  "94000",
  "94007",
  "94009",
  "94200",
  "94300",
  "94500",
  "94507",
  "94509",
  "94600",
  "94650",
  "94700",
  "94707",
  "94709",
  "94750",
  "94760",
  "94800",
  "94807",
  "94809",
  "94850",
  "94900",
  "94950",
  "95000",
  "95007",
  "95008",
  "95009",
  "95300",
  "95400",
  "95407",
  "95409",
  "95500",
  "95600",
  "95700",
  "95707",
  "95709",
  "95800",
  "95900",
  "96000",
  "96007",
  "96008",
  "96009",
  "96100",
  "96107",
  "96108",
  "96109",
  "96150",
  "96200",
  "96250",
  "96300",
  "96307",
  "96309",
  "96350",
  "96400",
  "96410",
  "96500",
  "96507",
  "96508",
  "96509",
  "96600",
  "96700",
  "96707",
  "96709",
  "96800",
  "96807",
  "96809",
  "96850",
  "96900",
  "96950",
  "97000",
  "97007",
  "97008",
  "97009",
  "97010",
  "97011",
  "97012",
  "97013",
  "97014",
  "97015",
  "97100",
]
`;

exports[`postcodejs:jest Func:postcode:Number 1`] = `
Object {
  "area": Array [
    "Kampung Bukit",
  ],
  "city": "Bintangor",
  "district": "Padang Terap",
  "state": "Sarawak",
}
`;

exports[`postcodejs:jest Func:postcode:String 1`] = `
Object {
  "area": Array [
    "Kampung Bukit",
  ],
  "city": "Bintangor",
  "district": "Padang Terap",
  "state": "Sarawak",
}
`;

exports[`postcodejs:jest Func:statePostcodes 1`] = `
Array [
  "93000",
  "93010",
  "93050",
  "93100",
  "93150",
  "93200",
  "93250",
  "93300",
  "93350",
  "93400",
  "93450",
  "93500",
  "93502",
  "93503",
  "93504",
  "93505",
  "93506",
  "93507",
  "93508",
  "93514",
  "93516",
  "93517",
  "93518",
  "93519",
  "93520",
  "93527",
  "93529",
  "93532",
  "93540",
  "93550",
  "93551",
  "93552",
  "93554",
  "93556",
  "93558",
  "93560",
  "93564",
  "93566",
  "93570",
  "93572",
  "93576",
  "93578",
  "93582",
  "93586",
  "93590",
  "93592",
  "93594",
  "93596",
  "93600",
  "93604",
  "93606",
  "93608",
  "93609",
  "93610",
  "93612",
  "93614",
  "93618",
  "93619",
  "93620",
  "93626",
  "93628",
  "93632",
  "93634",
  "93648",
  "93658",
  "93660",
  "93661",
  "93662",
  "93670",
  "93672",
  "93677",
  "93690",
  "93694",
  "93700",
  "93702",
  "93704",
  "93706",
  "93708",
  "93710",
  "93712",
  "93714",
  "93716",
  "93718",
  "93720",
  "93722",
  "93724",
  "93726",
  "93728",
  "93730",
  "93732",
  "93734",
  "93736",
  "93738",
  "93740",
  "93742",
  "93744",
  "93746",
  "93748",
  "93750",
  "93752",
  "93754",
  "93756",
  "93758",
  "93760",
  "93762",
  "93764",
  "93900",
  "93902",
  "93904",
  "93906",
  "93908",
  "93910",
  "93912",
  "93914",
  "93916",
  "93990",
  "94000",
  "94007",
  "94009",
  "94200",
  "94300",
  "94500",
  "94507",
  "94509",
  "94600",
  "94650",
  "94700",
  "94707",
  "94709",
  "94750",
  "94760",
  "94800",
  "94807",
  "94809",
  "94850",
  "94900",
  "94950",
  "95000",
  "95007",
  "95008",
  "95009",
  "95300",
  "95400",
  "95407",
  "95409",
  "95500",
  "95600",
  "95700",
  "95707",
  "95709",
  "95800",
  "95900",
  "96000",
  "96007",
  "96008",
  "96009",
  "96100",
  "96107",
  "96108",
  "96109",
  "96150",
  "96200",
  "96250",
  "96300",
  "96307",
  "96309",
  "96350",
  "96400",
  "96410",
  "96500",
  "96507",
  "96508",
  "96509",
  "96600",
  "96700",
  "96707",
  "96709",
  "96800",
  "96807",
  "96809",
  "96850",
  "96900",
  "96950",
  "97000",
  "97007",
  "97008",
  "97009",
  "97010",
  "97011",
  "97012",
  "97013",
  "97014",
  "97015",
  "97100",
  "97200",
  "97300",
  "98000",
  "98007",
  "98008",
  "98009",
  "98050",
  "98057",
  "98058",
  "98059",
  "98100",
  "98107",
  "98109",
  "98150",
  "98157",
  "98159",
  "98200",
  "98300",
  "98700",
  "98707",
  "98708",
  "98709",
  "98750",
  "98800",
  "98850",
  "98857",
  "98859",
]
`;
